src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -451,6 +494,41 @@ public void Close()
             }
         }
 
+        /// <summary>
+        /// Write the XDocument to the packages.config and disallow further changes.
+        /// </summary>
+        /// <param name="fullPath">the full path to packages.config file</param>
+        public void Close(string fullPath)
+        {
+            var directorypath = Path.GetDirectoryName(fullPath);
+
+            var tempConfigPath = Path.Combine(directorypath, Path.GetRandomFileName());
+
+            using (var tempConfigStream = File.Create(tempConfigPath))
+            {
+                // Save the XDocument to the temp stream
+                _xDocument.Save(tempConfigStream);
+            }
+
+            // Delete the existing packages.config file
+            if (File.Exists(fullPath))
```

Comment:
don't delete the file.

move it to packages.config.old.{datetime}

then do the move
then delete packages.config.old.{datetime}